Sources/MongoSwift/BSON/Document.swift

```diff
@@ -328,11 +349,7 @@ extension Document {
                 throw RuntimeError.internalError(message: toErrorString(error))
             }
 
-#if compiler(>=5.0)
```

Comment:
so `DocumentStorage(copying:)` (previously `DocumentStorage(fromPointer:)`) accepts a `BSONPointer` (non-mutable). 
previously, we were passing the result of `bson_new_from_json`, which is a `MutableBSONPointer`, into that initializer. On Swift 4.2, that required a cast to make the 
`UnsafeMutablePointer<bson_t>` into an `UnsafePointer<bson_t>`. 
on 5.0, since `BSONPointer` and `MutableBSONPointer` are both `OpaquePointer`s, no cast was necessary. hence the if/else.

since this code now calls the stealing initializer, which accepts a mutable pointer, no cast is necessary in 4.2.